### PR TITLE
Add BeforeUnload event support

### PR DESCRIPTION
**Pull request by Connormiha**
Add `beforeunload` event for `addEventListener`

https://developer.mozilla.org/en-US/docs/Web/Events/beforeunload

### CODE REVIEW

**Comment by mrkev** [comment]
Could you link to the spec that backs up this chagne?

**Comment by Connormiha** [comment]
Added link to mdn

File: lib/dom.js [hunk 1]
```diff
@@ -303,6 +310,10 @@ declare class FocusEvent extends UIEvent {
   relatedTarget: ?EventTarget;
 }
 
+declare class BeforeUnloadEvent extends Event {
+  returnValue: mixed;
```
Comment (nmote) [inline lib/dom.js]: According to the linked spec, it looks like this actually has type `string`. `mixed` is perfectly safe, but we might as well go with the more precise type here. Could you make that change, then I'll make sure it gets prompt attention?

**Comment by nmote** [comment]
Feel free to reopen if you would still like to get this in. For now I'm closing since I have not yet seen a response.

**Pull request closed without merging**